docs/reference/checks_config.md

```diff
@@ -0,0 +1,183 @@
+# Metadata Checks Configuration Schema
+
+This documentation explains the schema and configuration format for creating metadata validation rules in the checks system.
+
+## Overview
+
+The system provides a flexible way to define validation rules for record metadata using JSON stored in the database. Rules can check individual fields, compare values, and validate lists of items with complex conditions. This schema provides a powerful way to define complex validation logic while remaining human-readable and maintainable.
+
+## Rule Configuration
+
+A rule configuration is a dictionary with the following structure:
+
+```python
+{
+    "id": "unique-rule-id",  # Required
+    "title": "Human-readable title",
+    "message": "Error message shown when rule fails",
+    "description": "Detailed description of the rule",
+    "level": "info|warning|failure",  # Default: "info"
+    "condition": {  # Optional condition expression
+        "type": "expression_type",
+        # ... expression configuration
+    },
+    "checks": [  # List of check expressions
+        {
+            "type": "expression_type",
+            # ... expression configuration
+        }
+    ]
+}
+```
+
+This object is stored in the `params` section of the check_config in the database.
+
+## Expression Types
```

Comment:
I feel there's a small "prologue" section needed here to give some context about what an "expression" is, e.g. comparing it to Excel cell formulas or a scripting language "eval".